vars/neoDeploy.groovy

```diff
@@ -8,222 +11,218 @@ import com.sap.piper.tools.ToolDescriptor
 import groovy.transform.Field
 
 @Field String STEP_NAME = getClass().getName()
-@Field Set GENERAL_CONFIG_KEYS = []
-@Field Set STEP_CONFIG_KEYS = [
-    'account',
+@Field Set GENERAL_CONFIG_KEYS = [
+    'neo'
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
     'dockerEnvVars',
     'dockerImage',
     'dockerOptions',
-    'host',
-    'neoCredentialsId',
     'neoHome'
-]
+])
+
 @Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS.plus([
-    'applicationName',
-    'archivePath',
-    'deployAccount', //deprecated, replaced by parameter 'account'
-    'deployHost', //deprecated, replaced by parameter 'host'
+    'source',
     'deployMode',
-    'propertiesFile',
-    'runtime',
-    'runtimeVersion',
-    'vmSize',
     'warAction'
 ])
 
+@Field Map CONFIG_KEY_COMPATIBILITY = [
```

Comment:
Should we invest into compatibility when we announced already incompatible changes?  This is the opportunity to start at a new point without legacy. Of course it is polite to keep some fundamentals working from before, but this would be the chance to be really incompatible. And of course I know the story about having to find a balance between development efforts on the one hand and configuration efforts for the users on the other hand ...

Reply:
I am also fine with directly change it. However, I think have a time period where both works makes it easier to migrate to the new format. So you can already use the new version and convert the configuration afterwards.